app/models/ownership.rb

```diff
@@ -4,16 +4,7 @@ class Ownership < ActiveRecord::Base
 
   validates :user_id, uniqueness: { scope: :rubygem_id }
 
-  before_destroy :keep_last_owner
-
-  private
-
-  def keep_last_owner
-    if rubygem.owners.count == 1
-      errors[:base] << "Can't delete last owner of a gem."
-      false
-    else
-      true
-    end
+  def safe_destroy
+    rubygem.owners.count != 1 && destroy ? true : false
```

Comment:
`ownership` deletion is of two types. When profile gets deleted, [last owner of the gem gets deleted](https://github.com/rubygems/rubygems.org/pull/1301/files#diff-4676c008b11a5480d73d4a6de01e45b9R23) as well. However, when a user is removing an owner, we need to check that they are not removing the only owner. Callback would run on all the destroys, which is not what we need.

I know this is ugly but sadly this is the solution I have. Let me know if you can think of something better.